src/common/upload-flow.ts

```diff
@@ -49,6 +51,60 @@ export interface UploadFlowResult extends SynapseUploadResult {
   transactionHash?: string | undefined
 }
 
+/**
+ * Perform auto-funding if requested
+ * Automatically ensures a minimum of 10 days of runway based on current usage + new file requirements
+ *
+ * @param synapse - Initialized Synapse instance
+ * @param fileSize - Size of file being uploaded (in bytes)
+ * @param spinner - Optional spinner for progress
+ */
+export async function performAutoFunding(
+  synapse: Synapse,
+  fileSize: number,
+  spinner?: ReturnType<typeof import('../utils/cli-helpers.js').createSpinner>
```

Comment:
if we can avoid inline imports that would be preferable, I don't there's a good reason for this not to be a type import at the top of the file?